modules/core/src/main/java/org/apache/ignite/spi/discovery/tcp/messages/TcpDiscoveryStatusCheckMessage.java

```diff
@@ -118,7 +137,7 @@ public void status(int status) {
 
         TcpDiscoveryStatusCheckMessage other = (TcpDiscoveryStatusCheckMessage)obj;
 
-        return F.eqNodes(other.creatorNode, creatorNode) &&
+        return F.eqNodes(other.creatorNodeId(), creatorNodeId()) &&
```

Comment:
`F.eqNodes` cannot be used to check for equality of nodeIds; only instances of `ClusterNode` class are supported.